lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java

```diff
@@ -5314,4 +5320,16 @@ private void makeSamplePeerEval(UIContainer parent)
 		UIOutput.make(peerReviewRows, "peer-eval-sample-text", messageLocator.getMessage("simplepage.peer-eval.sample.4"));
 	}
 
+	private String personalizeText(String itemText) {
+		if (StringUtils.isNotBlank(itemText)) {
+			User currentUser = UserDirectoryService.getCurrentUser();
+			if (currentUser != null) {
```

Comment:
Hey Matthew, thanks for taking a look at this! Actually, ShowPageProducer still uses the deprecated UserDirectoryService static cover which _technically_ can return null, but I'm happy to remove the null check if you prefer. [UserDirectoryService.java#L136](https://github.com/sakaiproject/sakai/blob/master/kernel/api/src/main/java/org/sakaiproject/user/cover/UserDirectoryService.java#L136)

Do you have a preference for what should be printed if the user does not have a first, last, or full name? It probably makes the most sense to print a blank string for the name in those cases, but I can change it to whatever is preferred.

We actually do not use this feature anymore, but Dave Eveland requested it on the Jira ticket. So, I'm not invested at all in how it's implemented.